chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,325 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+        :param fee: A fee amount, taken out of the xch spend.
+        """
+        # get DID Wallet
+        for wallet_id in self.wallet_state_manager.wallets:
+            wallet = self.wallet_state_manager.wallets[wallet_id]
+            if wallet.type() == WalletType.DECENTRALIZED_ID.value:
+                if self.get_did() == hexstr_to_bytes(wallet.get_my_DID()):
+                    did_wallet = wallet
+                    break
+        else:
+            raise ValueError("There is no DID associated with this NFT wallet")
+
+        assert did_wallet.did_info.current_inner is not None
+        assert did_wallet.did_info.origin_coin is not None
+
+        # Ensure we have an edition_total value
+        if edition_total is None:
+            edition_total = len(metadata_list)
+        assert isinstance(edition_number_start, int)
+        assert len(metadata_list) <= edition_total + 1 - edition_number_start
+
+        # Ensure we have a did coin and its next inner puzzle hash
+        if did_coin is None:
+            coins = await did_wallet.select_coins(uint64(1))
+            assert coins is not None
+            did_coin = coins.pop()
+        innerpuz: Program = did_wallet.did_info.current_inner
+        if new_innerpuzhash is None:
+            new_innerpuzhash = innerpuz.get_tree_hash()
+        # make the primaries for the DID spend
+        primaries = [
+            AmountWithPuzzlehash(
+                {"puzzlehash": new_innerpuzhash, "amount": uint64(did_coin.amount), "memos": [new_innerpuzhash]}
+            )
+        ]
+
+        # Ensure we have an xch coin of high enough amount
+        assert isinstance(fee, uint64)
+        total_amount = len(metadata_list) + fee
+        if xch_coins is None:
+            xch_coins = await self.standard_wallet.select_coins(uint64(total_amount))
+        assert len(xch_coins) > 0
+
+        # set the chunk size for the spend bundle we're going to create
+        chunk_size = len(metadata_list)
+
+        # Because bulk minting may not mint all the NFTs in one bundle, we
+        # calculate the edition numbers that will be used in the intermediate
+        # puzzle based on the starting edition number given, and the size of the
+        # chunk going into this spend bundle
+        edition_number_end = edition_number_start + chunk_size
+
+        # Empty set to load with the announcements we will assert from DID to
+        # match the announcements from the intermediate launcher puzzle
+        did_announcements: Set = set()
+        puzzle_assertions: Set = set()
+        amount = uint64(1)
+        intermediate_coin_spends = []
+        launcher_spends = []
+        launcher_ids = []
+        eve_spends = []
+        transfer_spends = []
+        p2_inner_puzzle = await self.standard_wallet.get_new_puzzle()
+
+        # Loop to create each intermediate coin, launcher, eve and (optional) transfer spends
+        for edition_number in range(edition_number_start, edition_number_end):
```

Comment:
Naming this variable `edition_number` (and naming related method parameters `edition_number_start`, `edition_total`) is misleading as the actual edition number minted into the nft will be taken from `metadata_list` and may differ. And it rightly differs as explained in the next paragraph. I recommend to rename it (and related method parameters) to `mint_number`, `mint_list_offset` or similar.

Let me clarify again that many (certainly almost all existing) NFT collections will have `edition_number` and `edition_total` set to **1** for **each** NFT (Definition of "edition" as in [CHIP0005](https://github.com/Chia-Network/chips/blob/nft1/CHIPs/chip-0005.md). For example, Chia Friends would have edition_number set to 1 for each NFT as it does not contain "Identical copies of the same NFT" (CHIP 0005 definition of "edition".) I would be surprised if collections such as Chia Friends would not be able to benefit from a Bulk Mint tool (had to effectively single mint because their `edition_total` is 1.)